src/Microsoft.PowerShell.Commands.Utility/commands/utility/TestJsonCommand.cs

```diff
@@ -300,16 +292,27 @@ protected override void ProcessRecord()
 
         private void HandleValidationErrors(EvaluationResults evaluationResult)
         {
-            if (!evaluationResult.HasErrors)
+            if (evaluationResult.IsValid)
             {
                 return;
             }
 
-            foreach (var error in evaluationResult.Errors!)
+            if (evaluationResult.HasErrors)
```

Comment:
Please move the check to previous `if`.

Reply:
Removing this causes crashes. I tried it out in a test app, and the null forgive operator doesn't work on a nullable object being used for a foreach loop. I guess I remove the ! operator to make that more clear.

Reply:
Or maybe you meant we should combine evaluationResult.IsValid and evaluationResult.HasErrors? We can't do that, because for one we want to short-circuit any more evaluation, and for the other, there still maybe child results to visit.